tests/pipeline/test_pipeline_algo.py

```diff
@@ -738,3 +738,62 @@ def before_trading_start(context, data):
         )
 
         self.assertTrue(count[0] > 0)
+
+
+class PipelineSequenceTestCase(WithMakeAlgo, ZiplineTestCase):
+
+    def make_algo_kwargs(self, **overrides):
+        # run the algorithm for 3 days
+        self.START_DATE = pd.Timestamp('2015-07-13', tz='utc')
+        self.END_DATE = pd.Timestamp('2015-07-15', tz='utc')
+        return self.merge_with_inherited_algo_kwargs(
+            PipelineSequenceTestCase,
+            suite_overrides=dict(
+                get_pipeline_loader=lambda column: self.fake_pipeline,
+            ),
+            method_overrides=overrides,
+        )
+
+    def fake_pipeline(self):
+        raise AssertionError
+
+    def test_pipeline_compute_before_bts(self):
+
+        # for storing and keeping track of calls to BTS and TestFactor.compute
+        trace = []
+
+        class TestFactor(CustomFactor):
+            inputs = ()
+
+            # window_length doesn't actually matter for this test case
+            window_length = 1
+
+            def compute(self, today, assets, out):
+                trace.append("CustomFactor call")
+
+        def initialize(context):
+            pipeline = attach_pipeline(Pipeline(), 'my_pipeline')
+            test_factor = TestFactor()
+            pipeline.add(test_factor, 'test_factor')
+
+        def before_trading_start(context, data):
+            pipeline_output('my_pipeline')
+            trace.append("BTS call")
+
+        self.run_algorithm(
+            initialize=initialize,
+            before_trading_start=before_trading_start,
+            sim_params=SimulationParameters(
+                start_session=pd.Timestamp('2014-12-29', tz='utc'),
+                end_session=pd.Timestamp('2014-12-31', tz='utc'),
+                data_frequency='daily',
+                emission_rate='daily',
+                trading_calendar=self.trading_calendar,
+            )
+        )
+
+        # All pipeline computation calls should occur before any BTS calls,
+        # and the algorithm is being run for 3 days, so the first 3 calls
+        # should be to the custom factor and the next 3 calls should be to BTS
```

Comment:
Why is it three and then three and not interleaved?

Reply:
I believe this is because all three of the dates are included in the chunk, and therefore, they are all computed in one for loop -- see https://github.com/quantopian/zipline/blob/f61d3dfb141b12e14c9b4de332bdd7b7de620eb1/zipline/pipeline/mixins.py#L203